src/pokeService.js

```diff
@@ -17,6 +24,51 @@ async function getPokemonMoves(id) {
   return resolvePromises;
 }
 
+async function getPokemonEvoChain(id) {
+  const species = await getPokemonSpecies(id);
+  const evoChainUrl = species.data.evolution_chain.url;
+  const evoChains = await axios.get(evoChainUrl);
+  let evoData = evoChains.data.chain;
+
+  const evoChain = [];
+  do {
+    const numOfEvolutions = evoData.evolves_to.length;
+    let pokemon = await getPokemonByName(evoData.species.name);
+    evoChain.push({
+      url: evoData.species.url,
+      name: evoData.species.name,
+      id: pokemon.id,
+      image: pokemon.sprites.front_default,
+      min_level: !evoData.evolves_to[0] ? 1 : evoData.evolves_to[0].evolution_details[0].min_level,
+
+    });
+
+    if (numOfEvolutions > 1) {
+      for (let i = 1; i < numOfEvolutions; i++) {
+        pokemon = await getPokemonByName(evoData.evolves_to[i].species.name);
+        evoChain.push({
+          url: evoData.evolves_to[i].species.url,
+          name: evoData.evolves_to[i].species.name,
+          id: pokemon.id,
+          image: pokemon.sprites.front_default,
+          min_level: !evoData.evolves_to[i].evolution_details[0].min_level ? 1 : evoData.evolves_to[i].evolution_details[0].min_level
+        })
+        console.log(evoChain);
+      }
+    }
+    evoData = evoData["evolves_to"][0];
+  } while (!!evoData && evoData.hasOwnProperty("evolves_to"));
+
+  const resolvedPromises = await Promise.all(evoChain);
+  return resolvedPromises;
+}
+
+async function getPokemonSpecies(id) {
+  const url = `https://pokeapi.co/api/v2/pokemon-species/${id}`;
+  const species = await axios.get(url);
+  return species;
```

Comment:
There is a small inconsistency here. If you look at your get pokemon code, you're getting the result of the axios.get and then returning the data object inside it. You should be consistent on what you're returning on these functions. Either all return the `.data` or none do